### ISSUE TITLE
Broken display on Linux (v2)

### ISSUE THREAD

**Issue by acheong08** [wailsapp/wails#1471]
### Description

Init: Success

Build: Success

Dev: Page can be viewed in browser
```
main.js:41 TypeError: Cannot read properties of undefined (reading 'main')
    at Greet (App.js:6:22)
    at window.greet (main.js:32:9)
    at HTMLButtonElement.onclick ((index):15:2)
```

App: Displays a thin strip that cannot be viewed

### To Reproduce

1. `go install github.com/wailsapp/wails/v2/cmd/wails@latest`
2. Install all dependencies
3. `wails init -n myproject -t vanilla`
4. `cd myproject`
5. `wails build`
6. `wails dev`

### Expected behaviour

Create a working window

### Screenshots

![image](https://user-images.githubusercontent.com/36258159/174523597-d2a1b901-5fcd-4610-806e-f9bd19631a69.png)
![image](https://user-images.githubusercontent.com/36258159/174523723-de02b573-1b7d-45ec-926e-dde69d8fcc3b.png)

### Attempted Fixes

Run X11

### System Details

```shell
$ wails doctor                 
Wails CLI v2.0.0-beta.37


Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:		Ubuntu
Version: 	22.04
ID:		ubuntu
Go Version:	go1.18.1
Platform:	linux
Architecture:	amd64

Wails
------
Version: 		v2.0.0-beta.37
Package Manager: 	apt

Dependency	Package Name		Status		Version
----------	------------		------		-------
*docker		docker.io		Installed	20.10.17
gcc		build-essential		Installed	12.9ubuntu3
libgtk-3	libgtk-3-dev		Installed	3.24.33-1ubuntu2
libwebkit	libwebkit2gtk-4.0-dev	Installed	2.36.3-0ubuntu0.22.04.1
npm		npm			Installed	8.5.1~ds-1
*nsis		nsis			Installed	v3.08-2
pkg-config	pkg-config		Installed	0.29.2-1ubuntu3

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
```
```


### Additional context

Custom kernel

**Comment by acheong08** [comment]
Wails version 1 is working fine though

**Comment by leaanthony** [comment]
Is this a dupe of #1459?

**Comment by acheong08** [comment]
Yes and no. In #1459 the app works. The main issue here is that the app doesn't show up

**Comment by acheong08** [comment]
This could be related to #1420 as well seeing as Ubuntu 22.04 no longer supports X11.

**Comment by leaanthony** [comment]
Have you tried this: 
https://github.com/wailsapp/wails/issues/1420#issuecomment-1162932318

**Comment by acheong08** [comment]
I'm checking if GTK 4 fixes this issue

**Comment by acheong08** [comment]
Unlike #1420 where `Gtk-WARNING **: 15:42:30.500: cannot open display:`, the display is found and rendered. However, there is an issue with rendering which results on only a very thin piece of the display showing up.

**Comment by ianmjones** [comment]
This is strange, I'm able to `wails dev` and/or build & run on Ubuntu 22.04 with v2.0.0-beta.37.

Although, I had a previous issue (#1207) that exhibited similar problems when trying to go fullscreen on ARM.

Does your app change window dimensions on start up or do anything else with the window?

**Comment by leaanthony** [comment]
It sounds like the bindings don't get injected into the frontend based on the error. Maybe a library version issue?

**Comment by leaanthony** [comment]
It sounds like the bindings don't get injected into the frontend based on the error. Maybe a library version issue?

**Comment by acheong08** [comment]
> This is strange, I'm able to `wails dev` and/or build & run on Ubuntu 22.04 with v2.0.0-beta.37.
> 
> Although, I had a previous issue (#1207) that exhibited similar problems when trying to go fullscreen on ARM.
> 
> Does your app change window dimensions on start up or do anything else with the window?

What versions of dependencies are you using?

**Comment by acheong08** [comment]
The window is unable to be manipulated

**Comment by leaanthony** [comment]
@ianmjones - thanks for jumping on this. Do your gtk/webkit library versions align with those reported here?

**Comment by ianmjones** [comment]
Looks to be exactly the same, I just don't have docker or nsis installed:

```
System
------
OS:		Ubuntu
Version: 	22.04
ID:		ubuntu
Go Version:	go1.18.1
Platform:	linux
Architecture:	amd64

Wails
------
Version: 		v2.0.0-beta.37
Package Manager: 	apt

Dependency	Package Name		Status		Version
----------	------------		------		-------
*docker		docker.io		Available	20.10.12-0ubuntu4
gcc		build-essential		Installed	12.9ubuntu3
libgtk-3	libgtk-3-dev		Installed	3.24.33-1ubuntu2
libwebkit	libwebkit2gtk-4.0-dev	Installed	2.36.3-0ubuntu0.22.04.1
npm		npm			Installed	8.5.5
*nsis		nsis			Available	3.08-2
pkg-config	pkg-config		Installed	0.29.2-1ubuntu3

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - docker: sudo apt install docker.io
  - nsis: sudo apt install nsis
```

App runs fine after build (not shown, wails dev is fine too):

![image](https://user-images.githubusercontent.com/4710/175809345-326b4f42-c5b6-4110-bbd3-8c9b993da727.png)

**Comment by acheong08** [comment]
Closing in favor of #1517
